torchmdnet/datasets/md17.py

```diff
@@ -1,105 +1,151 @@
+        'revised naphthalene': 'rmd17_naphthalene.npz',
+        'revised aspirin': 'rmd17_aspirin.npz',
+        'revised salicylic acid': 'rmd17_salicylic.npz',
+        'revised malonaldehyde': 'rmd17_malonaldehyde.npz',
+        'revised ethanol': 'rmd17_ethanol.npz',
+        'revised toluene': 'rmd17_toluene.npz',
+        'revised paracetamol': 'rmd17_paracetamol.npz',
+        'revised azobenzene': 'rmd17_azobenzene.npz',
+        'benzene CCSD(T)': 'benzene_ccsd_t.zip',
+        'aspirin CCSD': 'aspirin_ccsd.zip',
+        'malonaldehyde CCSD(T)': 'malonaldehyde_ccsd_t.zip',
+        'ethanol CCSD(T)': 'ethanol_ccsd_t.zip',
+        'toluene CCSD(T)': 'toluene_ccsd_t.zip',
+        'benzene FHI-aims': 'benzene2018_dft.npz',
+    }
+
+    def __init__(
+        self,
+        root: str,
+        molecules: str,
+        train: Optional[bool] = None,
+        transform: Optional[Callable] = None,
+        pre_transform: Optional[Callable] = None,
+        pre_filter: Optional[Callable] = None,
+    ):
+        name = molecules
+        if name not in self.file_names:
+            raise ValueError(f"Unknown dataset name '{name}'")
+
+        self.name = name
+        self.revised = 'revised' in name
+        self.ccsd = 'CCSD' in self.name
+
+        super().__init__(root, transform, pre_transform, pre_filter)
+
+        if len(self.processed_file_names) == 1 and train is not None:
+            raise ValueError(
+                f"'{self.name}' dataset does not provide pre-defined splits "
+                f"but the 'train' argument is set to '{train}'")
+        elif len(self.processed_file_names) == 2 and train is None:
+            raise ValueError(
+                f"'{self.name}' dataset does provide pre-defined splits but "
+                f"the 'train' argument was not specified")
+
+        idx = 0 if train is None or train else 1
+        self.data, self.slices = torch.load(self.processed_paths[idx])
+
+    def mean(self) -> float:
+        return float(self._data.energy.mean())
 
     @property
-    def processed_file_names(self):
-        return [f"md17-{mol}.pt" for mol in self.molecules]
+    def raw_dir(self) -> str:
+        if self.revised:
+            return osp.join(self.root, 'raw')
+        return osp.join(self.root, self.name, 'raw')
 
-    def download(self):
-        for file_name in self.raw_file_names:
-            download_url(MD17.raw_url + file_name, self.raw_dir)
-
-    def process(self):
-        for raw_path, processed_path in zip(self.raw_paths, self.processed_paths):
-            data_npz = np.load(raw_path)
-            z = torch.from_numpy(data_npz["z"]).long()
-            positions = torch.from_numpy(data_npz["R"]).float()
-            energies = torch.from_numpy(data_npz["E"]).float()
-            forces = torch.from_numpy(data_npz["F"]).float()
+    @property
+    def processed_dir(self) -> str:
+        return osp.join(self.root, self.name, 'processed')
```

Comment:
Now using this, you put your processed data in: self.root/self.name/processed. 
I don't know if this could help: `return osp.join(self.root, 'processed', self.name)` it's going to create a processed dir for your specific data in data/processed so you can keep the conventional data gerarchy: self.root/raw and self.root/processed.